pkg/client/client.go

```diff
@@ -245,6 +403,71 @@ func (c *ImmuClient) Reference(keyReader io.Reader, valueReader io.Reader) (*sch
 	return result, err
 }
 
+// SafeReference ...
+func (c *ImmuClient) SafeReference(keyReader io.Reader, valueReader io.Reader) (*VerifiedIndex, error) {
+	start := time.Now()
+	if !c.isConnected() {
+		return nil, ErrNotConnected
+	}
+	key, err := ioutil.ReadAll(valueReader)
+	if err != nil {
+		return nil, err
+	}
+	reference, err := ioutil.ReadAll(keyReader)
+	if err != nil {
+		return nil, err
+	}
+
+	rs := NewRootService(c.serviceClient, cache.NewFileCache())
+	root, err := rs.GetRoot(context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &schema.SafeReferenceOptions{
+		Ro: &schema.ReferenceOptions{
+			Reference: &schema.Key{Key: reference},
```

Comment:
In the future We should semplify this model inside .proto file and use a more simple
```
message ReferenceOptions {
	bytes reference = 1;
	bytes key = 2;
}
```